unittest/extra.cpp

```diff
@@ -36,8 +36,31 @@
 #include "gamedata.h"
 
 /// Run the initial setup for a faction
-/// which for a unit test is to do nothing
-int Game::SetupFaction( Faction *pFac ) { return 1; }
+/// For unit tests, factions get 1 leader with combat 3.
```

Comment:
In order to unit test json reports, it still needs a valid game object correctly set up, so I construct a unit test world which has exactly 4 hexes and 1 city.  The test will also have guards and wandering monsters, so those 2 factions are taken up as well.